rootfs/ingest/python/qserv/queue.py

```diff
@@ -249,30 +250,33 @@ def _run_lock_queries(self, contribfiles_to_lock_count: int) -> int:
             Number of contribfiles locked
         """
 
-        self._wait_for_mutex()
+        try:
+            self._acquire_mutex()
 
-        select_query = select([self.queue.c.id])
-        select_query = select_query.limit(contribfiles_to_lock_count)
-        select_query = select_query.where(self.queue.c.locking_pod.is_(None))
-        select_query = select_query.where(self.queue.c.database == self.contribution_metadata.database)
+            select_query = select([self.queue.c.id])
+            select_query = select_query.limit(contribfiles_to_lock_count)
+            select_query = select_query.where(self.queue.c.locking_pod.is_(None))
+            select_query = select_query.where(self.queue.c.database == self.contribution_metadata.database)
 
-        # _LOG.debug("Query select: %s", select_query.compile(dialect=mysql.dialect()))
-        with self.engine.connect() as connection:
-            rows = connection.execute(select_query)
-            ids = []
-            for e in rows:
-                ids.append(e[0])
-            rows.close()
-
-        update_query = update(self.queue).values(
-            locking_pod=self.pod
-        )
-        update_query = update_query.where(self.queue.c.id.in_(ids))
-
-        # _LOG.debug("Query (MySQL dialect)): %s", update_query.compile(dialect=mysql.dialect()))
-        self._safe_execute(update_query, _MAX_RETRY_ATTEMPTS)
-
-        self._release_mutex()
+            # _LOG.debug("Query select: %s", select_query.compile(dialect=mysql.dialect()))
+            with self.engine.connect() as connection:
+                rows = connection.execute(select_query)
+                ids = []
+                for e in rows:
+                    ids.append(e[0])
+                rows.close()
+
+            update_query = update(self.queue).values(
+                locking_pod=self.pod
+            )
+            update_query = update_query.where(self.queue.c.id.in_(ids))
+
+            # _LOG.debug("Query (MySQL dialect)): %s", update_query.compile(dialect=mysql.dialect()))
+            self._safe_execute(update_query, _MAX_RETRY_ATTEMPTS)
+        except Exception as e:
+            raise IngestError("Failed to lock contribution files in queue") from e
+        finally:
+            self._release_mutex()
```

Comment:
It's probably better to use context manager instead of try/finally.

Reply:
Interesting, I add it to my TODO list because this is not blocking!